src/cleanvision/dataset/folder_dataset.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from typing import List, Optional, Dict
+
+import pandas as pd
+from PIL import Image
+
+from cleanvision.dataset import Dataset
+from cleanvision.utils.utils import get_filepaths
+
+
+class FolderDataset(Dataset):
+    def __init__(
+        self, data_folder: Optional[str] = None, filepaths: Optional[List[str]] = None
+    ) -> None:
+        super().__init__()
+        if data_folder:
+            self._filepaths = get_filepaths(data_folder)
+        else:
+            assert filepaths is not None
+            self._filepaths = filepaths
+        self._set_index()
+        self.metadata["index_to_path"] = pd.DataFrame(
+            {"image_path": self._filepaths}, index=self.index
+        )
+
+    def __len__(self) -> int:
+        return len(self._filepaths)
+
+    def __getitem__(self, item: int) -> Image.Image:
+        path = self._filepaths[item]
+        return Image.open(path)
+
+    def _set_index(self) -> None:
+        self.index = [i for i in range(len(self._filepaths))]
```

Comment:
consider as str instead of int. Just set `index: Union[int, str]` in the parent class if you have LSP problems